src/sign.c

```diff
@@ -373,30 +373,35 @@ int wally_ec_sig_verify(const unsigned char *pub_key, size_t pub_key_len,
                         const unsigned char *sig, size_t sig_len)
 {
     secp256k1_pubkey pub;
+    secp256k1_xonly_pubkey xonly_pub;
     secp256k1_ecdsa_signature sig_secp;
     const secp256k1_context *ctx = secp_ctx();
     bool ok;
 
-    if (!pub_key || pub_key_len != EC_PUBLIC_KEY_LEN ||
-        !bytes || bytes_len != EC_MESSAGE_HASH_LEN ||
+    if (!pub_key || (pub_key_len != EC_PUBLIC_KEY_LEN &&
+        pub_key_len != EC_XONLY_PUBLIC_KEY_LEN))
+        return WALLY_EINVAL;
+
+    if (!bytes || bytes_len != EC_MESSAGE_HASH_LEN ||
         !is_valid_ec_type(flags) || flags & ~EC_FLAGS_TYPES ||
         !sig || sig_len != EC_SIGNATURE_LEN)
         return WALLY_EINVAL;
 
     if (!ctx)
         return WALLY_ENOMEM;
 
-    ok = pubkey_parse(&pub, pub_key, pub_key_len);
-
-    if (flags & EC_FLAG_SCHNORR)
-#if 0 /*FIXME: Schnorr is unavailable in secp for now*/
-        ok = ok && secp256k1_schnorr_verify(ctx, sig, bytes, &pub);
-#else
-        ok = false;
-#endif
-    else
+    if (flags & EC_FLAG_SCHNORR) {
+        if (pub_key_len != EC_XONLY_PUBLIC_KEY_LEN)
+            return WALLY_EINVAL;
+		ok = xpubkey_parse(&xonly_pub, pub_key, pub_key_len);
```

Comment:
- Misleading indenting.
-  `xpubkey_parse` will take either a full or an xonly pubkey, so there is no harm in removing the length check and allowing the caller to verify with a normal compressed pubkey IMO. The fewer places that force the distinction on the caller externally, the better.